src/libs/actions/Subscription.ts

```diff
@@ -61,4 +63,64 @@ function updateSubscriptionType(type: SubscriptionType) {
     });
 }
 
-export {openSubscriptionPage, updateSubscriptionType};
+function updateSubscriptionSize(newSubscriptionSize: number, currentSubscriptionSize: number) {
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: ONYXKEYS.NVP_PRIVATE_SUBSCRIPTION,
+                value: {
+                    userCount: newSubscriptionSize,
+                    errorFields: {
+                        userCount: null,
+                    },
+                    pendingFields: {
+                        userCount: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: ONYXKEYS.NVP_PRIVATE_SUBSCRIPTION,
+                value: {
+                    userCount: newSubscriptionSize,
+                    errorFields: {
+                        userCount: null,
+                    },
+                    pendingFields: {
+                        userCount: null,
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: ONYXKEYS.NVP_PRIVATE_SUBSCRIPTION,
+                value: {
+                    userCount: currentSubscriptionSize,
+                    errorFields: {
+                        userCount: ErrorUtils.getMicroSecondOnyxError('common.genericErrorMessage'),
+                    },
```

Comment:
```suggestion
```

Let's not add the generic error message in the frontend data. Forcing an error from my backend, this is the result:


https://github.com/Expensify/App/assets/7277067/5b35e667-1d0a-4938-84d4-f58ff1bc3073

Because the microtimes are different, both errors display which is more confusing than helpful.

If we remove the generic error we achieve the desired result:

https://github.com/Expensify/App/assets/7277067/f3ae23da-239f-4ab0-9627-0d08caa36346

Also **NOTE**: I need to make a small backend update to send the errors in `['errorFields', 'userCount']` (currently it is sending it in `errors` key, but I agree with your implementation that we need to send it in `['errorFields', 'userCount']`, otherwise we can't distinguish where exactly on the Subscription screen to display the error). cc @blimpich, I'm assigning you for review on that PR

Reply:
The `errorFields` backend change is now on staging by the way

Reply:
Sure thing, changing it